java/src/jmri/jmrit/beantable/SensorTableAction.java

```diff
@@ -317,23 +317,27 @@ protected void setDefaultState(JFrame _who) {
     public void setMenuBar(BeanTableFrame f) {
         final jmri.util.JmriJFrame finalF = f;			// needed for anonymous ActionListener class
         JMenuBar menuBar = f.getJMenuBar();
-
-        JMenu optionsMenu = new JMenu(Bundle.getMessage("MenuDefaults"));
-        JMenuItem item = new JMenuItem(Bundle.getMessage("GlobalDebounce"));
-        optionsMenu.add(item);
-        item.addActionListener(new ActionListener() {
-            public void actionPerformed(ActionEvent e) {
-                setDefaultDebounce(finalF);
-            }
-        });
-        item = new JMenuItem(Bundle.getMessage("InitialSensorState"));
-        optionsMenu.add(item);
-        item.addActionListener(new ActionListener() {
-            public void actionPerformed(ActionEvent e) {
-                setDefaultState(finalF);
-            }
-        });
-        menuBar.add(optionsMenu);
+        // check for menu
+        int menus = menuBar.getMenuCount();
+        if (menus < 3) {
```

Comment:
Instead of counting the menus to find where you want to insert the menu, iterate through the menu items, checking against the text of each menu item to determine if what you want to add has not already been added.